wbobeirne/joule-extension
# Add modal to suggest fixes for connection failures

Closes #121 

### Description

This PR implements a quick fix to the issue where a user temporarily accepts the LND self-signed certificate via their browser. After some time, the browser will begin rejecting this certificate and the user needs to approve it again.

### Steps to Test

1. Connect to your node using the LND self-signed certificate
2. Wait for the browser to eventually reject the cert, this may take days 😞 . I have not found a way to induce this behavior. I tested using other browsers (Brave, Firefox, Opera) that already had timed-out certs.
3. Open Joule and you should see the connection error modal on the home screen
4. Click on the `clicking this link` link to open a new tab with the certificate error screen
5. Click on "Advanced" then proceed to accept the certificate
6. Reopen Joule or if it was open in Full Page mode, switch back to the tab then click the `Retry` button. The connection should work properly now
7. The error modal also suggests re-uploading macaroons if the certificate is not the problem


### Screenshots

![image](https://user-images.githubusercontent.com/1356600/54036996-0f08b800-418b-11e9-9899-cc3fe89ddac7.png)

![image](https://user-images.githubusercontent.com/1356600/54037013-1e880100-418b-11e9-8e41-0e56d3dce89e.png)
